airbyte-integrations/connectors/source-salesforce/source_salesforce/api.py

```diff
@@ -0,0 +1,348 @@
+    "BatchApexErrorEvent",
+    "LoginEventStream",
+    "ReportEventStream",
+    "AIPredictionEvent",
+    "SessionHijackingEvent",
+    "PlatformStatusAlertEvent",
+    "ApiEventStream",
+    "LightningUriEventStream",
+    "LogoutEventStream",
+    "FlowExecutionErrorEvent",
+    "AsyncOperationEvent",
+    "BulkApiResultEvent",
+    "LoginAsEventStream",
+    "ConcurLongRunApexErrEvent",
+    "ApiAnomalyEvent",
+    "UriEventStream",
+    "ProcessExceptionEvent",
+    "ListViewEventStream",
+    "ReportAnomalyEvent",
+    "AsyncOperationStatus",
+]
+
+
+UNSUPPORTED_BULK_API_SALESFORCE_OBJECTS = [
+    "Attachment",
+    "CaseStatus",
+    "ContractStatus",
+    "DeclinedEventRelation",
+    "FieldSecurityClassification",
+    "OrderStatus",
+    "PartnerRole",
+    "RecentlyViewed",
+    "ServiceAppointmentStatus",
+    "SolutionStatus",
+    "TaskPriority",
+    "TaskStatus",
+    "UndecidedEventRelation",
+    "AcceptedEventRelation",
+    "AssetTokenEvent",
+    "AttachedContentNote",
+    "EventWhoRelation",
+    "QuoteTemplateRichTextData",
+    "TaskWhoRelation",
+]
+
+
+UNSUPPORTED_FILTERING_STREAMS = [
+    "ApiEvent",
+    "BulkApiResultEventStore",
+    "EmbeddedServiceDetail",
+    "EmbeddedServiceLabel",
+    "FormulaFunction",
+    "FormulaFunctionAllowedType",
+    "FormulaFunctionCategory",
+    "IdentityProviderEventStore",
+    "IdentityVerificationEvent",
+    "LightningUriEvent",
+    "ListViewEvent",
+    "LoginAsEvent",
+    "LoginEvent",
+    "LogoutEvent",
+    "Publisher",
+    "RecordActionHistory",
+    "ReportEvent",
+    "TabDefinition",
+    "UriEvent",
+]
+
+
+class Salesforce:
+    version = "v52.0"
+
+    def __init__(
+        self,
+        refresh_token: str = None,
+        token: str = None,
+        client_id: str = None,
+        client_secret: str = None,
+        is_sandbox: bool = None,
+        start_date: str = None,
+        api_type: str = None,
+    ):
+        self.api_type = api_type.upper() if api_type else None
+        self.refresh_token = refresh_token
+        self.token = token
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.access_token = None
+        self.instance_url = None
+        self.session = requests.Session()
+        self.is_sandbox = is_sandbox is True or (isinstance(is_sandbox, str) and is_sandbox.lower() == "true")
+        self.start_date = start_date
+
+    def _get_standard_headers(self):
+        return {"Authorization": "Bearer {}".format(self.access_token)}
+
+    def get_streams_black_list(self) -> List[str]:
+        black_list = QUERY_RESTRICTED_SALESFORCE_OBJECTS + QUERY_INCOMPATIBLE_SALESFORCE_OBJECTS
+        if self.api_type == "REST":
+            return black_list
+        else:
+            return black_list + UNSUPPORTED_BULK_API_SALESFORCE_OBJECTS
+
+    def filter_streams(self, stream_name: str) -> bool:
+        if stream_name.endswith("ChangeEvent") or stream_name in self.get_streams_black_list():
```

Comment:
why ignore change event?

Reply:
REST and BULK API  do not support all entities that end with ChangeEvent for receiving data on them.

I`ll add comment